src/GraphCoarsenBuilder.cpp

```diff
@@ -190,17 +210,255 @@ void AssembleMBuilder::AddTraceAcross(int row, int col, double value)
+                                          const mfem::SparseMatrix& Pedges,
+                                          const mfem::SparseMatrix& face_cdof)
+{
+    // in future MFEM releases when SparseMatrix::GetSubMatrix is const-correct,
+    // the next line will no longer be necessary
+    mfem::SparseMatrix& Pedges_noconst = const_cast<mfem::SparseMatrix&>(Pedges);
+
+    // F_F block
+    const int num_faces = topology_.Agg_face_.Width();
+    const int num_aggs = topology_.Agg_face_.Height();
+    mfem::Array<int> local_fine_dofs;
+    mfem::Array<int> local_coarse_dofs;
+    mfem::Vector local_fine_weight;
+    comp_F_F_.resize(num_faces);
+    for (int face = 0; face < num_faces; ++face)
+    {
+        GetCoarseFaceDofs(face_cdof, face, local_coarse_dofs);
+        GetTableRowCopy(topology_.face_edge_, face, local_fine_dofs);
+        fineMdiag.GetSubVector(local_fine_dofs, local_fine_weight);
+        mfem::DenseMatrix P_F(local_fine_dofs.Size(), local_coarse_dofs.Size());
+        Pedges_noconst.GetSubMatrix(local_fine_dofs, local_coarse_dofs, P_F);
+        comp_F_F_[face] = RTDP(P_F, local_fine_weight, P_F);
+    }
+
+    // the EF_EF block
+    // for (pairs of *faces* that share an *aggregate*)
+    mfem::Array<int> local_faces;
+    mfem::Array<int> local_fine_dofs_prime;
+    mfem::Array<int> local_coarse_dofs_prime;
+    for (int agg = 0; agg < num_aggs; ++agg)
+    {
+        GetTableRowCopy(topology_.Agg_face_, agg, local_faces);
+        GetTableRowCopy(topology_.Agg_edge_, agg, local_fine_dofs);
+        fineMdiag.GetSubVector(local_fine_dofs, local_fine_weight);
+        for (int f = 0; f < local_faces.Size(); ++f)
+        {
+            int face = local_faces[f];
+            GetCoarseFaceDofs(face_cdof, face, local_coarse_dofs);
+            mfem::DenseMatrix P_EF(local_fine_dofs.Size(), local_coarse_dofs.Size());
+            Pedges_noconst.GetSubMatrix(local_fine_dofs, local_coarse_dofs, P_EF);
+            for (int fprime = f; fprime < local_faces.Size(); ++fprime)
+            {
+                int faceprime = local_faces[fprime];
+                // GetTableRowCopy(topology_.face_edge_, faceprime, local_fine_dofs_prime);
+                GetCoarseFaceDofs(face_cdof, faceprime, local_coarse_dofs_prime);
+                mfem::DenseMatrix P_EFprime(local_fine_dofs.Size(), local_coarse_dofs_prime.Size());
+                Pedges_noconst.GetSubMatrix(local_fine_dofs, local_coarse_dofs_prime, P_EFprime);
+                comp_EF_EF_.push_back(RTDP(P_EF, local_fine_weight, P_EFprime));
+            }
+        }
+    }
+
+    // EF_E block and E_E block
+    comp_E_E_.resize(num_aggs);
+    for (int agg = 0; agg < num_aggs; ++agg)
+    {
+        GetCoarseAggDofs(agg, local_coarse_dofs);
+        if (local_coarse_dofs.Size() == 0)
+        {
+            mfem::DenseMatrix empty(0, 0);
+            comp_E_E_[agg] = empty;
+            for (int af = 0; af < local_faces.Size(); ++af)
```

Comment:
This loop's stop condition `local_faces.Size()` is not set for the current aggregate.  It will use whatever size was previously set.  The test cases don't cover this scenario.  
There is probably a much simpler set of parameters to make this crash, but the following  segfaults on my machine
```sh
./generalgraph -m 4  -t 0.01 --coarse-coefficient -gg -nv 5000 -np 1000 -gf -ma
```

Reply:
Thanks, good catch, I've fixed these two bugs.